src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,26 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        StringBuilder data = new StringBuilder();
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(fileName));
+            for (String s : lines) {
+                data.append(s);
+            }
+        } catch (IOException e) {
+            throw new FileNotFoundException();
+        }
+        char[] chars = data.toString().toLowerCase().replaceAll("[^a-z]","").toCharArray();
+        Set<Character> characterSet = new TreeSet<>();
+        for (char ch : chars) {
+            characterSet.add(ch);
+        }
+        StringBuilder builder = new StringBuilder();
+        for (char ch : characterSet) {
+            builder.append(ch);
+        }
+
+        return builder.length() < 5 ? builder.toString() : builder.toString().substring(0, 5);
```

Comment:
5 is a magic number here. I suggest to make it constant.

Reply:
I agree

Reply:
done